### ISSUE TITLE
net: TestDialerDualStackFDLeak failures due to "i/o timeout" on solaris-amd64-oraclerel builder

### ISSUE THREAD

**Issue by bcmills** [golang/go#43070]
[2020-10-30T14:13:03-733c4af/solaris-amd64-oraclerel](https://build.golang.org/log/a43748a6ed42dbc2b149e66d9d91b5051c5f946c)
[2020-10-29T13:53:33-c45d780/solaris-amd64-oraclerel](https://build.golang.org/log/977d105ec0722ed2e149e4e0330072305b41cdb6)
[2020-10-28T19:28:08-87d59bc/solaris-amd64-oraclerel](https://build.golang.org/log/74cc1fc185c7147fb429294d497b591ede34d9f2)
[2020-10-28T00:48:22-49b017f/solaris-amd64-oraclerel](https://build.golang.org/log/87a2f1b00e9aa29a842f88d6734ef12adfac1649)

**Comment by bcmills** [comment]
This is probably related to #41629.

**Comment by bcmills** [comment]
CC @rorth @fraenkel @bradfitz @ianlancetaylor

**Comment by rorth** [comment]
As an initial test, I've built trunk on an identical system and ran
net.test in a tight loop of

	net.test -test.run TestDialerDualStackFDLeak

On an unloaded system, I got no failure at all in 30000+ runs.  I've
than repeated the test on the host of the builder (under some load due
to 3 different buildbots total running there): here I find 7 FAILs out
of ca. 68000 runs.  Adding some load on the initial system, I get some
FAILs, too (2 out of 60000).

The question is how best to proceed from here.
